### ISSUE TITLE
Groupby Array-Type Quantiles Broken in 0.25.0

### ISSUE THREAD

**Issue by sernst** [pandas-dev/pandas#27526]
#### Code Sample

```python
import pandas as pd

df = pd.DataFrame({
    'category': ['A', 'A', 'A', 'A', 'A', 'A', 'B', 'B', 'B', 'B', 'B', 'B'],
    'value': [1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6]
})
quantiles = df.groupby('category').quantile([0.25, 0.5, 0.75])
print(quantiles)
```
#### Problem description

In previous versions of Pandas `< 0.25.0`  and in the documentation it is possible to pass an array-type of quantiles into the `DataFrameGroupBy.quantile()` method to return multiple quantile values in a single call. However, upon installation of `0.25.0` the following error results instead:

```
Traceback (most recent call last):
  File "example.py", line 8, in <module>
    quantiles = df.groupby('category').quantile([0.25, 0.5, 0.75])
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 1908, in quantile
    interpolation=interpolation,
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 2248, in _get_cythonized_result
    func(**kwargs)  # Call func to modify indexer values in place
  File "pandas/_libs/groupby.pyx", line 69
```

#### Expected Output

Using Pandas `0.24.2` the output is:

```
               value
category
A        0.25   2.25
         0.50   3.50
         0.75   4.75
B        0.25   2.25
         0.50   3.50
         0.75   4.75
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.9.125-linuxkit
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : None
pytest           : 5.0.1
hypothesis       : None
sphinx           : 2.1.2
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : None
pandas_datareader: None
bs4              : 4.8.0
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : 2.6.9
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : 0.3.0
scipy            : 1.3.0
sqlalchemy       : None
tables           : 3.5.2
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

**Comment by pilkibun** [comment]
Confirmed as regression. Before #20405, `GroupBy.quantile` dispatches to `DataFrame.quantile` which still accepts a list. 

https://github.com/pydata/pandas/blob/76247c142893c710e970c4cf8a25d73121aa5a2b/pandas/core/groupby/groupby.py#L557-L564

https://github.com/pydata/pandas/blob/76247c142893c710e970c4cf8a25d73121aa5a2b/pandas/core/groupby/groupby.py#L596-L599

after #20405, this now goes to a dedicated cython function which only accepts a single quantile value
https://github.com/pydata/pandas/blob/76247c142893c710e970c4cf8a25d73121aa5a2b/pandas/_libs/groupby.pyx#L688-L694

**Comment by ar4hc** [comment]
I got this error  message when using a numpy array (form np.linspace()):

    TypeError: only size-1 arrays can be converted to Python scalars

Downgrade to pandas 0.24 solves this.

my test code (snipplet):

        percs = (np.linspace(0, 1, num=intervals + 1).round(decimals=3))
        d = df[['x', 'y']]
        g = d.groupby('x')
        quants = g.quantile(percs)

breaks in last line with 0.25, works in 0.24

**Comment by jreback** [comment]
there is a PR #27473 which solves this and just needs some touching up to fix

**Comment by pilkibun** [comment]
That PR was about #20405 not validating inputs. This issue is about #20405 deleting functionality so different bugs.

**Comment by TomAugspurger** [comment]
Is the fix to change https://github.com/pandas-dev/pandas/blob/c0ff67a22df9c18da1172766e313732ed2ab6c30/pandas/core/groupby/groupby.py#L1900-L1911 to be called once per value in `q`, when a list of quintiles is provide? Then concat the results together with `concat(results, axis=1, keys=q)`?

**Comment by TomAugspurger** [comment]
The output of `DataFrameGroupBy.quantile` is a DataFrame whose 

* index is the group keys
* columns are the (numeric) columns

```python
In [68]: df = pd.DataFrame({"A": [0, 1, 2, 3, 4]})

In [69]: df.groupby([0, 0, 1, 1, 1]).quantile(0.25)
Out[69]:
      A
0  0.25
```

What's the expected output of `.quantile(List[float])`?

It's not the most useful, but I think the best option is a MultiIndex in the columns.

```python
In [70]: a = df.iloc[:2].quantile([0.25]).unstack()

In [71]: b = df.iloc[2:].quantile([0.25]).unstack()

In [72]: pd.concat([a, b], keys=[0, 1]).unstack([1, 2])
Out[72]:
      A
   0.25
0  0.25
1  2.50

```

The other option is to have the `q`s in the index, but that breaks my mental model that the index should be the unique group keys.

**Comment by TomAugspurger** [comment]
Oh, whoops, I missed the 0.24 output. We'll match that.

**Comment by dragoljub** [comment]
Thanks for the fix. I just ran into this!

**Comment by ar4hc** [comment]
not sure if this is the right place,but with 0.25.1 and my code from above i now get a differnet error, but still an error:

```
    quants = g.quantile(percs)
  File "/usr/local/miniconda3/envs/scipy37/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 1953, in quantile
    return result.take(indices)
  File "/usr/local/miniconda3/envs/scipy37/lib/python3.7/site-packages/pandas/core/generic.py", line 3604, in take
    indices, axis=self._get_block_manager_axis(axis), verify=True
  File "/usr/local/miniconda3/envs/scipy37/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 1389, in take
    indexer = maybe_convert_indices(indexer, n)
  File "/usr/local/miniconda3/envs/scipy37/lib/python3.7/site-packages/pandas/core/indexers.py", line 201, in maybe_convert_indices
    raise IndexError("indices are out-of-bounds")
IndexError: indices are out-of-bounds
```

the dataframe groupyby objects is looking good, the percs ars a list of floats, including 0 and 1, and the data is the same as it was for the for 0.24.2 version.

what am i missing....?

**Comment by TomAugspurger** [comment]
A new issue with a minimal example would be welcome.
http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

**Comment by TomAugspurger** [comment]
See https://github.com/pandas-dev/pandas/pull/28113

**Comment by ar4hc** [comment]
sry, just reverted back to 0.24 and went to fix the other issues i have... :-\ 
and didn't watch here.

**Comment by dangoldner** [comment]
Ran into this in 1.0.1: 
```
import numpy as np
import pandas as pd
dates=pd.date_range(start='2000-01-01',end='2000-01-03',periods=25).tz_localize('utc')
group = np.random.randint(4,size=25); 
df = pd.DataFrame({'group':group,'dates':dates})
df.groupby('group').quantile(.3)
```
output: 
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-72-942190bf95c9> in <module>
      2 group = np.random.randint(4,size=25);
      3 df = pd.DataFrame({'group':group,'dates':dates})
----> 4 df.groupby('group').quantile(.3)

~/anaconda3/envs/py38/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in quantile(self, q, interpolation)
   1909 
   1910         if is_scalar(q):
-> 1911             return self._get_cythonized_result(
   1912                 "group_quantile",
   1913                 aggregate=True,

~/anaconda3/envs/py38/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in _get_cythonized_result(self, how, cython_dtype, aggregate, needs_values, needs_mask, needs_ngroups, result_is_index, pre_processing, post_processing, **kwargs)
   2296                 func = partial(func, ngroups)
   2297 
-> 2298             func(**kwargs)  # Call func to modify indexer values in place
   2299 
   2300             if result_is_index:

pandas/_libs/groupby.pyx in pandas._libs.groupby.__pyx_fused_cpdef()

TypeError: No matching signature found
```
The error doesn't occur if `.tz_localize` isn't called so `dates` remains tz-naive.
